### PR TITLE
Generate the localeinfo.json files directly

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- many differences from ilib, but they are minor and/or they are corrections (we need to investigate why the ilib files are wrong in those cases)
- generator scripts are now in es6 code, and much cleaner and smaller than the ilib cldr tools

### CODE REVIEW

File: test/testLocaleInfo.js [hunk 1]
```diff
@@ -12820,8 +12820,8 @@ module.exports.testlocaleinfo = {
         var info = new LocaleInfo("ar-AE");
         test.ok(info !== null);
 
-        test.equal(info.getWeekEndStart(), 5);
-        test.equal(info.getWeekEndEnd(), 6);
+        test.equal(info.getWeekEndStart(), 6);
+        test.equal(info.getWeekEndEnd(), 0);
```
Comment (gouniLee) [inline test/testLocaleInfo.js:12824]: When I checked cldr 41, For AE country, weeEndStart is Friday, and weekendEnd is Saturday. so modified test case is wrong.

File: test/testLocaleInfo.js [hunk 2]
```diff
@@ -6116,7 +6116,7 @@ module.exports.testlocaleinfo = {
         var info = new LocaleInfo("ks-Arab-IN");
         test.ok(info !== null);
 
-        test.equal(info.getCurrencyFormats().common, "{s} {n}");
+        test.equal(info.getCurrencyFormats().common, "{s}{n}");
```
Comment (gouniLee) [inline test/testLocaleInfo.js:6119]: seems it is cldr changed from cldr 40 to 41. Probbally, tcs in iLib need to udpate.

File: scripts/calendars.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (gouniLee) [inline scripts/calendars.js:1]: you updated lots for cldr locale data scripts, but tcs are only for localeinfo.
I think It needs to make sure all updated scripts are not broken in current related test cases.

File: test/testLocaleInfo.js (hunk 1 above)
Comment (ehoogerbeets) [reply test/testLocaleInfo.js:12824]: Where did you check?

Take a look: https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/weekData.json

AE is not listed in the weekend start and end lists...

Also, a snippet from CLDR 41 XML in `common/supplemental/supplementalData.xml`:

```
               <firstDay day="sun" territories="GB" alt="variant" references="Shorter Oxford Dictionary (5th edition, 2002)"/>
                <weekendStart day="thu" territories="AF"/>
                <weekendStart day="fri" territories="BH DZ EG IL IQ IR JO KW LY OM QA SA SD SY YE"/>
                <weekendStart day="sat" territories="001"/>
                <weekendStart day="sun" territories="IN UG"/>
                <weekendEnd day="fri" territories="AF IR"/>
                <weekendEnd day="sat" territories="BH DZ EG IL IQ JO KW LY OM QA SA SD SY YE"/>
                <weekendEnd day="sun" territories="001"/>
```

That seems to have the same information as the json above.

File: test/testLocaleInfo.js (hunk 2 above)
Comment (ehoogerbeets) [reply test/testLocaleInfo.js:6119]: I submitted PRs for cldr 41 a few months back in the ilib project, remember?

What does "tcs" stand for?

Comment (gouniLee) [reply test/testLocaleInfo.js:6119]: I meant test case here. : https://github.com/iLib-js/iLib/blob/development/js/test/root/testlocaleinfo.js#L6211
I think iLib haven't updated all feature to cldr41. I saw only some feature update PRs. no?

File: test/testLocaleInfo.js (hunk 1 above)
Comment (gouniLee) [reply test/testLocaleInfo.js:12824]: oh sorry. I checked the wrong file.. you are right.
Not all features of iLib were not updated to cldr 41, and due to that, there are test cases expectation is different. that's why I'm confused.

File: scripts/calendars.js (hunk 3 above)
Comment (gouniLee) [reply scripts/calendars.js:1]: I thought some scripts aren't related localeinfo.json file generation. I realized I misunderstood. all scrips files for generating localeinfo.json.

**Comment by ehoogerbeets** [comment]
Okay, ready to re-review

**Comment by gouniLee** [comment]
facing an error below while running the script file.
ilib-localeinfo/scripts$ node genlocinfo.js
```
internal/fs/utils.js:308
    throw err;
    ^

Error: ENOENT: no such file or directory, open '../tz/zone.tab'
    at Object.openSync (fs.js:476:3)
    at Object.readFileSync (fs.js:377:35)
    at new UnicodeFile (/home/goun/Source/opensource_iLib/ilib-localeinfo/node_modules/ilib-data-utils/src/UnicodeFile.js:64:27)
    at setUpTimeZones (/home/goun/Source/opensource_iLib/ilib-localeinfo/scripts/timezones.js:51:21)
    at genTimeZones (/home/goun/Source/opensource_iLib/ilib-localeinfo/scripts/timezones.js:106:5)
    at Object.<anonymous> (/home/goun/Source/opensource_iLib/ilib-localeinfo/scripts/locinfo.js:55:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Module._compile (/home/goun/Source/opensource_iLib/ilib-localeinfo/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Object.newLoader [as .js] (/home/goun/Source/opensource_iLib/ilib-localeinfo/node_modules/pirates/lib/index.js:141:7) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '../tz/zone.tab'
}

```

**Comment by ehoogerbeets** [comment]
Please run `npm run build:tz` first, and then `npm run build:data`

**Review by gouniLee — APPROVED** [review]
Don't forget to check the test expectations difference between here and iLib itself.

**Pull request merged**